working/1426-extension-types/feature-specification.md

```diff
@@ -0,0 +1,948 @@
+`E`, and with an unnamed single argument constructor setting that field to
+the argument. A getter `E.class get box` is implicitly induced in `E`, and
+it returns an object that wraps `this`.
+
+`E.class` also implicitly induces a getter `E get unbox` which returns the
+value of the final field mentioned above, typed as the associated extension
+type.
+
+*In the case where the extension type is protected, the `unbox` getter
+cannot be written in Dart (because a cast to `E` is then a compile-time
+error), so `unbox` must be a language feature in this case. For
+convenience, it is induced implicitly for all extension types.*
+
+In the case where it would be a compile-time error to declare such a member
+named `box` or `unbox`, said member is not induced.
+
+*The latter rule helps avoiding conflicts in situations where `box` or
+`unbox` is a non-hidden instance member, and it allows developers to write
+their own implementations if needed.*
+
+*The rationale for having this mechanism is that the wrapper object is a
+full-fledged object: It is a subtype of all the direct superinterfaces of
+`E`, so it can be used in code where `E` is not in scope. Moreover, it
+supports late binding of the extension methods, and even dynamic
+invocations. It is costly (it takes space and time to allocate and
+initialize the wrapper object), but it is more robust than the extension
+type, which will only work in a manner which is resolved statically.*
+
+
+## Dynamic Semantics
+
+The dynamic semantics of extension method invocation is very similar to
+the semantics of invocations of the existing extension methods:
+
+If `e` is an expression whose static type is the extension type
+<code>Ext<S<sub>1</sub>, .. S<sub>k</sub>></code>,
+then a member access like `e.m(args)` is executed by evaluating `e` to an
+object which is bound to a fresh variable `v`, and then evaluating
+<code>Ext<S<sub>1</sub>, .. S<sub>k</sub>>(v).m(args)</code>,
+and similarly for instance getters and operators.
+
+The dynamic semantics of an invocation of an instance method of the on-type
+which is enabled in an explicit extension type by the show/hide part is as
+if a forwarder were implicitly induced in the extension, with the same
+signature as that of the on-type. *For example:*
+
+```dart
+// Extension type using show/hide:
+extension type MyNum on num show floor {}
+
+// Works like the following:
+extension type MyNum on num {
+  int floor() => this.floor();
+}
+```
+
+*Note that this implies that the extension method `floor` never overrides
+the instance member `floor`, but the extension method `floor` will be
+executed at a call site `myNum.floor()` based on a compile-time decision
+when the receiver `myNum` has static type `MyNum`. In particular, the
+extension method `floor` will never be executed when the receiver has type
+`dynamic`. The forwarding expression `this.floor()` in the implicitly
+induced getter will invoke the instance method, which is subject to late
```

Comment:
What getter are you referring to? There are not getters that I can see in this example.

Reply:
Right, should have been 'method'. Done.